File: API/Entities/ContactPerson.cs

using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace API.Entities
{
    public class ContactPerson
    {
        public int Id { get; set; }
        [Required]
        public string Name { get; set; }
        public string Position { get; set; }
        public string PhoneNumber { get; set; }
        [EmailAddress]
        public string Email { get; set; }

        [Required]
        public int SupplierId { get; set; }  // Foreign key to Supplier
        [JsonIgnore]
        public Supplier Supplier { get; set; }

    }
}

File: client/src/models/contactPerson.ts
import { Supplier } from './supplier'

export interface ContactPerson {
    id: number
    name: string
    position: string
    phoneNumber: string
    email: string
    supplier: Supplier

}
export interface ContactPersonDto {
    id: number
    name: string
    position: string
    phoneNumber: string
    email: string
    supplierId: number

}
// Define the initial empty supplier object
export const emptyContactPeron: ContactPersonDto = {
    id: 0,
    name: '',
    position: '',
    email: '',
    phoneNumber: '',
    supplierId: 0


}